src/Catalog/PackageCatalogItem.cs

```diff
@@ -94,6 +103,52 @@ public override IGraph CreateContentGraph(CatalogContext context)
             //  identity and version
             SetIdVersionFromGraph(graph);
 
+            // deprecation
+            if (Deprecation != null)
+            {
+                // assert deprecation root node to subject
+                var deprecationPredicate = graph.CreateUriNode(Schema.Predicates.Deprecation);
+                var deprecationRootNode = graph.CreateUriNode(new Uri(resource.Subject.ToString() + "#deprecation"));
+                graph.Assert(resource.Subject, deprecationPredicate, deprecationRootNode);
+
+                // assert reasons to deprecation root node
+                var deprecationReasonRootNode = graph.CreateUriNode(Schema.Predicates.DeprecationReasons);
+                foreach (var reason in Deprecation.Reasons)
+                {
+                    var reasonNode = graph.CreateLiteralNode(reason);
```

Comment:
Can the reason be null?

Reply:
nope, and `Deprecation.Reasons` throws if it's null or empty.